File: bookstore-micronaut-example/bookstore-service/src/main/java/example/bookstore/micronaut/persistence/repository/BookRepository.java
package example.bookstore.micronaut.persistence.repository;


import example.bookstore.micronaut.persistence.entity.BookEntity;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.jpa.repository.JpaSpecificationExecutor;
import io.micronaut.data.repository.PageableRepository;

@Repository
public interface BookRepository extends PageableRepository<BookEntity, Long>, JpaSpecificationExecutor<BookEntity> {
}

File: bookstore-service-micronaut-kotlin/src/main/kotlin/com/example/micronaut/bookstore/repository/BookRepository.kt
package com.example.micronaut.bookstore.repository

import com.example.micronaut.bookstore.repository.entity.BookEntity
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaSpecificationExecutor
import io.micronaut.data.repository.CrudRepository

@Repository
interface BookRepository : CrudRepository<BookEntity, String>, JpaSpecificationExecutor<BookEntity>